packages/cli/commands/project/cloneApp.js

```diff
@@ -0,0 +1,177 @@
+const fs = require('fs');
+const path = require('path');
+const {
+  addAccountOptions,
+  addConfigOptions,
+  getAccountId,
+  addUseEnvironmentOptions,
+} = require('../../lib/commonOpts');
+const { trackCommandUsage } = require('../../lib/usageTracking');
+const { loadAndValidateOptions } = require('../../lib/validation');
+const { i18n } = require('../../lib/lang');
+const {
+  selectPublicAppPrompt,
+} = require('../../lib/prompts/selectPublicAppPrompt');
+const { promptUser } = require('../../lib/prompts/promptUtils');
+const { poll } = require('../../lib/polling');
+const {
+  uiLine,
+  uiCommandReference,
+  uiAccountDescription,
+} = require('../../lib/ui');
+const SpinniesManager = require('../../lib/ui/SpinniesManager');
+const {
+  logApiErrorInstance,
+  ApiErrorContext,
+} = require('../../lib/errorHandlers/apiErrors');
+const { EXIT_CODES } = require('../../lib/enums/exitCodes');
+const { isAppDeveloperAccount } = require('../../lib/accountTypes');
+const {
+  cloneApp,
+  checkCloneStatus,
+  downloadClonedProject,
+} = require('@hubspot/local-dev-lib/api/projects');
+const { getCwd, isValidPath } = require('@hubspot/local-dev-lib/path');
+const { logger } = require('@hubspot/local-dev-lib/logger');
+const { getAccountConfig } = require('@hubspot/local-dev-lib/config');
+const {
+  fetchPublicAppMetadata,
+} = require('@hubspot/local-dev-lib/api/appsDev');
+const { extractZipArchive } = require('@hubspot/local-dev-lib/archive');
+
+const i18nKey = 'commands.project.subcommands.cloneApp';
+
+exports.command = 'clone-app [--appId] [--location]';
+exports.describe = null; // uiBetaTag(i18n(`${i18nKey}.describe`), false);
+
+exports.handler = async options => {
+  await loadAndValidateOptions(options);
+
+  const accountId = getAccountId(options);
+  const accountConfig = getAccountConfig(accountId);
+  const accountName = uiAccountDescription(accountId);
+
+  trackCommandUsage('clone-app', {}, accountId);
+
+  if (!isAppDeveloperAccount(accountConfig)) {
+    uiLine();
+    logger.error(i18n(`${i18nKey}.errors.invalidAccountTypeTitle`));
+    logger.log(
+      i18n(`${i18nKey}.errors.invalidAccountTypeDescription`, {
+        useCommand: uiCommandReference('hs accounts use'),
+        authCommand: uiCommandReference('hs auth'),
+      })
+    );
+    uiLine();
+    process.exit(EXIT_CODES.ERROR);
```

Comment:
We're not the most consistent in how we use these exit codes, but I think we want this to be `SUCCESS`, since nothing is actually going wrong here. That's how I thought about it for local dev stuff at least. Might be something we should try to make some more concrete rules for

cc @brandenrodgers @joe-yeager

Reply:
Addressed in https://github.com/HubSpot/hubspot-cli/pull/1085/commits/247b941649560ed561734257bca012ddf2d65f02